users/shared_account.go

```diff
@@ -0,0 +1,230 @@
+			"You can connect to your account to manage it : https://re.trackit.io/")
+		err := mail.SendMail(userMail, mailSubject, mailBody, ctx)
+		if err != nil {
+			logger.Error("Failed to send email.", err.Error())
+			return err
+		}
+	} else {
+		dbForgottenPassword, token, err := resetPasswordGenerator(ctx, tx, newUserId)
+		mailSubject := "You are invited to join Trackit"
+		mailBody := fmt.Sprintf("Hi, you have been invited to join trackit. Please follow this link to create" +
+			" your account: https://re.trackit.io/reset/%d/%s.", dbForgottenPassword.ID, token)
+		err = mail.SendMail(userMail, mailSubject, mailBody, ctx)
+		if err != nil {
+			logger.Error("Failed to send viewer password email.", err.Error())
+			return err
+		}
+	}
+	return nil
+}
+
+// inviteUserWithValidBody tries to share an account with a specific user
+func inviteUserWithValidBody(request *http.Request, body inviteUserRequest, tx *sql.Tx, user User) (int, interface{}) {
+	logger := jsonlog.LoggerFromContextOrDefault(request.Context())
+	result, guestId, err := checkUserWithEmail(request.Context(), tx, body.Email)
+	if err == nil {
+		if result {
+			isAlreadyShared, err := checkSharedAccount(request.Context(), tx, body.AccountId, guestId)
+			if err != nil {
+				return 403, errors.New("An error occured while inviting a user. Please try again.")
+			} else if isAlreadyShared {
+				return 200, errors.New("You are already sharing this account with this user.")
+			}
+			err = addAccountToGuest(request.Context(), tx, body.AccountId, body.PermissionLevel, guestId)
+			if err == nil {
+				err = sendMailNotification(request.Context(), tx, body.Email,true, 0)
+				if err != nil {
+					logger.Error("Error occured while sending an email to an existing user.", err.Error())
+					return 403, errors.New("An error occured while inviting a user. Please, try again.")
+				}
+				return 200, nil
+			} else {
+				logger.Error("Error occured while adding account to an existing user.", err.Error())
+				return 403, errors.New("An error occured while inviting a user. Please, try again.")
+			}
+		} else {
+			newUserId, err := createAccountForGuest(request.Context(), tx, body.Email, body.AccountId, body.PermissionLevel)
+			if err == nil {
+				err = sendMailNotification(request.Context(), tx, body.Email,false, newUserId)
+				if err != nil {
+					logger.Error("Error occured while sending an email to a new user.", err.Error())
+					return 403, errors.New("An error occured while inviting a new user. Please, try again.")
+				}
+				return 200, nil
+			} else {
+				logger.Error("Error occured while creating new account for a guest.", err.Error())
+				return 403, errors.New("An error occured while inviting a new user. Please, try again.")
+			}
+		}
+	} else {
+		logger.Error("Error occured while checking if user already exist.", err.Error())
+		return 403, errors.New("An error occured while inviting a new user. Please, try again.")
+	}
+}
```

Comment:
I see a lot of errors, you should pack them in the top of the file
```
var (
   ...
)
```
(see `users/auth.go`)